packages/ui/src/components/Modal/Disclosure.tsx

```diff
@@ -0,0 +1,28 @@
+import styled from '@emotion/styled'
+import type { DisclosureProps } from './types'
+
+const DisclosureContainer = styled.button`
+  display: contents;
+`
+
+export const Disclosure = ({
+  disclosure,
+  handleOpen,
+  visible,
+  handleClose,
+  toggle,
+  id,
+}: DisclosureProps) =>
+  typeof disclosure === 'function' ? (
+    disclosure({
+      visible,
+      onClose: handleClose,
+      toggle,
+      onOpen: handleOpen,
+      modalId: id,
+    })
+  ) : (
+    <DisclosureContainer onClick={handleOpen} type="button" aria-controls={id}>
+      {disclosure}
+    </DisclosureContainer>
```

Comment:
There is an issue with this, we are creating a button wrapper around the disclosure which is also a button. The result is 2 buttons in each others in the dom:

![Screenshot 2023-08-30 at 11 23 28](https://github.com/scaleway/ultraviolet/assets/15812968/688233c3-2b1d-4069-a7a9-fea38ab1abff)